src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
This can be removed, right?

Also, may test directory mtime first, and return `false` if greater than passed timestamp.